### DISCUSSION TITLE
Builds both modules and APKs

### DISCUSSION THREAD

**Discussion by oldman20** [nikhilbadyal/docker-py-revanced#174] [Ideas]
Can you development version support Builds both modules and APKs? Thankfully @nikhilbadyal 
I knew it available in your repo here https://github.com/nikhilbadyal/revanced-magisk-module
but i want it has same feature like upload to telegram automatically like this repo

**Comment by nikhilbadyal** [comment]
Use [this](https://github.com/nikhilbadyal/revanced-magisk-module/tree/extended) or [this](https://github.com/j-hc/revanced-magisk-module) for modules.

**Reply by oldman20** [reply]
thank, i successed with it, but it doesnt have upload to  Telegram funtion like this repo

**Reply by nikhilbadyal** [reply]
This repo also provides an option where you can mention any other (2nd) repo and it will upload whatever is available on second repo to the telegram channel. 

Actually that's how I maintain my own channel. I build magisk module there(2nd repo) and then ask this repo [(here)](https://github.com/nikhilbadyal/docker-py-revanced/blob/dd09a50951e7f7e569747ce3fd23ecd104846bd0/.github/workflows/build-apk.yml#L130) to upload its contenet(2nd repo) to telegram.

You need to provide **DOWNLOAD_GITHUB_REPOSITORY** env which point to 2nd repo.

**Reply by oldman20** [reply]
> You need to provide **DOWNLOAD_GITHUB_REPOSITORY** env which point to 2nd repo.

In **Repository secret**?

**Reply by nikhilbadyal** [reply]
yes

**Reply by oldman20** [reply]
> yes

please provide a sample for it?! i dont know syntax

**Reply by oldman20** [reply]
> This repo also provides an option where you can mention any other (2nd) repo and it will upload whatever is available on second repo to the telegram channel.
> 
> Actually that's how I maintain my own channel. I build magisk module there(2nd repo) and then ask this repo [(here)](https://github.com/nikhilbadyal/docker-py-revanced/blob/dd09a50951e7f7e569747ce3fd23ecd104846bd0/.github/workflows/build-apk.yml#L130) to upload its contenet(2nd repo) to telegram.
> 
> You need to provide **DOWNLOAD_GITHUB_REPOSITORY** env which point to 2nd repo.

How  can i use other repo to upload?
![image](https://github.com/nikhilbadyal/docker-py-revanced/assets/39668756/5244b51e-a434-40ec-8809-019b73dcd9e6)

**Reply by oldman20** [reply]
In ENVS i put this is correct?
DOWNLOAD_GITHUB_REPOSITORY=https://github.com/j-hc/revanced-magisk-module

**Reply by nikhilbadyal** [reply]
Dont put in ENVS. Create new secret **DOWNLOAD_GITHUB_REPOSITORY**. This is how my secret looks like(see 4th)
<img width="777" alt="image" src="https://github.com/nikhilbadyal/docker-py-revanced/assets/59223300/869d641e-512c-4d99-9203-86d140603549">

**Reply by oldman20** [reply]
Thank, but what content format i put in that value? Link url of repo?

**Reply by nikhilbadyal** [reply]
Just the onwer and repo name.Seperated by **_/_** . So it can be 

**oldman20/revanced-magisk-module**

**Reply by oldman20** [reply]
Thank you so much 😀, got it
Edit: i updated it but still show current repo in Action>Run workflow
![image](https://github.com/nikhilbadyal/docker-py-revanced/assets/39668756/113ad41b-8466-468c-841e-a6ff7260f422)

**Reply by oldman20** [reply]
but it work, just repo upload name in workflow still not change

**Reply by nikhilbadyal** [reply]
It not possible to change it dynamically (Github doesn't allow that) and hence is hardcoded. 
If you want to change it, Do it  [here](https://github.com/nikhilbadyal/docker-py-revanced/blob/dd09a50951e7f7e569747ce3fd23ecd104846bd0/.github/workflows/build-apk.yml#L20)

**Reply by oldman20** [reply]
Great help. Thank so much!
